crates/rerun_c/src/rerun.h

```diff
@@ -0,0 +1,106 @@
+// The Rerun C SDK.
+
+#ifndef RERUN_H
+#define RERUN_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+// ----------------------------------------------------------------------------
+// Types:
+
+#define RERUN_STORE_KIND_RECORDING 1
+#define RERUN_STORE_KIND_BLUEPRINT 2
+
+/// What is returned by the first call to `rerun_rec_stream_new`.
+/// Usually you only have one recording stream, so you can call
+/// `rerun_rec_stream_new` once, ignore its return value, and use
+/// `RERUN_REC_STREAM_DEFAULT` everywhere in your code.
+#define RERUN_REC_STREAM_DEFAULT 0
+
+/// A unique handle for a recording stream.
+///
+/// The default is RERUN_REC_STREAM_DEFAULT
+typedef int32_t RerunRecStream;
```

Comment:
Why `RecStream` rather than `RecordingStream`? That adds confusion with very little gains imo

Reply:
Changed it to `rr_recording_stream`, to make the entire C SDK follow `rr_snake_case`. Feels more C:ish.

I'll use `PascalCase` for types in the C++ SDK though when we get there.